### PR TITLE
 feat: add image from path and image to response

### PR DESCRIPTION
**Pull request by pudility**
The following code:
```JavaScript
const resp = await fetch(pictureURL)
const original = new fly.Image(await resp.arrayBuffer())

let image = original.resize(800).withoutEnlargement()

image = await image.toImage()
return new Response(image.data, resp)
```
Can now be turned into:
```JavaScript
  const original = await fly.Image.imageFromPath(pictureURL)

  let image = original.resize(800).withoutEnlargement()

  return await image.toResponse()
```

### CODE REVIEW

**Comment by mrkurt** [comment]
I need to think about this, I don't think I like the API but I'm not sure what would be better. In theory, `new Image(url)` would be best but that might be tough since loading URLs is async.

File: v8env/ts/fly/image.ts [hunk 1]
```diff
@@ -146,6 +146,18 @@ export class Image {
     i.info = result.info
     return i
   }
+
+  async toResponse(res:any = { headers: { "Content-Type": "image/jpg" }}): Promise<any> {
+    const result = await imageToBuffer(this._ref)
```
Comment (mrkurt) [inline v8env/ts/fly/image.ts]: This should just call `this.toBuffer()` instead of reimplementing the logic here.

Comment (pudility) [reply v8env/ts/fly/image.ts]: I will change that.

**Comment by pudility** [comment]
@mrkurt I agree, `new Image(url)` would be better. I spent a solid amount of time trying to make this work but, because constructors can't be async - it makes it very difficult (there would probably have to be a blocking loop which is very ugly). 

Another option that might work would be adding another class or a module that extends the image class - I don't know whats best - something to think about.

**Comment by pudility** [comment]
Does anything else need to be done on this?

**Comment by jeromegn** [comment]
@pudility @mrkurt with URL argument in the constructor, it could lazily fetch the resource if/when necessary to apply operations on it later.

However, we'd have to figure out what type of argument we got in the constructor and change the behaviour. A bit annoying but doable.

Also worth noting: the Web API prescribes `new Image(width, height)` (arguments are optional). This is a `fly` off-spec thing, so it's fine (and also out of the scope of this PR.)

**Pull request closed without merging**